### PR TITLE
feature(@teller-network) licenses can be bought only with SNT

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: contracts/teller-network/License.sol [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    event Bought(address buyer, uint256 price);
+    event RecipientChanged(address _recipient);
+    event PriceChanged(uint256 _price);
+    event Released(address buyer);
+    event ReleaseDelayChanged(uint256 _newDelay);
+
+    uint256 public reserveAmount;
+
+ 
+
```
Comment (jrainville) [inline contracts/teller-network/License.sol]: Lots of spaces here

File: contracts/teller-network/License.sol [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    constructor(address payable _tokenAddress, address payable _recipient, uint256 _price, uint256 _releaseDelay) public {
+        recipient = _recipient;
+        price = _price;
+        releaseDelay = _releaseDelay;
+        reserveAmount = 0;
+        token = ERC20Token(_tokenAddress);
+    }
+
+    /**
+    * @dev Check is the address already owns a license
```
Comment (jrainville) [inline contracts/teller-network/License.sol]:
```suggestion
    * @dev Check if the address already owns a license
```
small typo

File: contracts/teller-network/License.sol [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+     * @notice Withdraw not reserved tokens
+     * @param _token Address of ERC20 withdrawing excess, or address(0) if want ETH.
+     * @param _beneficiary Address to send the funds.
+     **/
+    function withdrawExcessBalance(
+        address _token,
+        address payable _beneficiary
+    )
+        external 
+        onlyOwner
```
Comment (jrainville) [inline contracts/teller-network/License.sol]: Why put this on multiple lines?

Comment (jrainville) [reply contracts/teller-network/License.sol]: Same for the others

File: contracts/token/ERC20Receiver.sol [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    function _depositToken(
+        address _from,
+        ERC20Token _token,
+        uint256 _amount
+    )
+        private 
+    {
+        require(_amount > 0, "Bad argument");
```
Comment (jrainville) [inline contracts/token/ERC20Receiver.sol]: Just for my information, we deposit negative amounts of tokens?

Comment (richard-ramos) [reply contracts/token/ERC20Receiver.sol]: @alaibe , good idea for changing the text to be more explicit.

**Pull request merged**